runtime/vm/ContinuationHelpers.cpp

```diff
@@ -198,8 +223,19 @@ yieldContinuation(J9VMThread *currentThread)
 	 *
 	 * must be maintained for weakly ordered CPUs, to unsure that once the continuation is again available for GC scan (on potentially remote CPUs), all CPUs see up-to-date stack .
 	 */
-	Assert_VM_true((uintptr_t)currentThread == continuation->state);
-	continuation->state = J9_GC_CONTINUATION_STATE_CONCURRENT_SCAN_NONE;
+	Assert_VM_true(VM_VMHelpers::isContinuationMountedWithCarrierThread(continuation->state, currentThread));
```

Comment:
this assert should be just before resetContinuationCarrierID or even inside it (in which case we would pass current thread)

Reply:
ok, not possible to move inside, so let's move it just in front of reset